py-polars/polars/internals/frame.py

```diff
@@ -463,16 +463,15 @@ def read_csv(
             if isinstance(file, StringIO):
                 file = file.getvalue().encode()
 
-        projection: Optional[List[int]] = None
+        projection: Optional[Sequence[int]] = None
         if columns:
-            if isinstance(columns, list):
```

Comment:
Note that I have removed these checks for list, that is now happening inside `is_***_sequence` methods. It effectively does change the behaviour when a non-None value is passed in that is not a list, but the type annotations do not allow for that to start with, so I felt safe doing this. Idem for the other methods.